compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -272,26 +262,39 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
     }
 }
 
-struct BuildReducedGraphVisitor<'a, 'b, 'tcx> {
-    r: &'b mut Resolver<'a, 'tcx>,
-    parent_scope: ParentScope<'a>,
+impl<'a, 'tcx> ExpandResolver<'a, 'tcx> {
+    pub(crate) fn build_reduced_graph(
+        &mut self,
+        fragment: &AstFragment,
+        parent_scope: ParentScope<'a>,
+    ) -> MacroRulesScopeRef<'a> {
+        collect_definitions(&mut self.r, fragment, parent_scope.expansion);
+        let mut visitor = BuildReducedGraphVisitor { r: self, parent_scope };
+        fragment.visit_with(&mut visitor);
+        visitor.parent_scope.macro_rules
+    }
+}
+
+pub(crate) struct BuildReducedGraphVisitor<'a, 'b, 'tcx> {
```

Comment:
Does `BuildReducedGraphVisitor` actually use `ExpandResolver`?
From what I understand it should never use the `containers_deriving_copy` and therefore can use `Resolver`.